daemon/commit.go

```diff
@@ -74,6 +74,7 @@ func merge(userConf, imageConf *containertypes.Config) error {
 	if len(userConf.Entrypoint) == 0 {
 		if len(userConf.Cmd) == 0 {
 			userConf.Cmd = imageConf.Cmd
+			userConf.ArgsEscaped = imageConf.ArgsEscaped
```

Comment:
At some point yes, this is the right direction to go in and something I'm actively looking at doing in the future - have a proof of concept of that, but it's non trivial unfortunately.